io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -1007,7 +1009,8 @@ impl LvsLvol for Lvol {
     async fn destroy_replica(mut self) -> Result<String, Error> {
         let snapshot_lvol = self.is_snapshot_clone();
         let name = self.name();
-        self.destroy().await?;
+        self.clone().destroy().await?;
```

Comment:
hmm not sure if it's safe to clone here as we're deleting it, we might then later access memory which is now freed?
Might be better to put event on var first, and then after delete here then just call generate?

Reply:
Followed this now.

Also changed replica creation and deletion events to when "lvol" is created and deleted.